File: bot/supabase_config.py
#!/usr/bin/env python3
"""
GTM Supabase Configuration
Конфигурация для интеграции с Supabase
"""

import os
from dotenv import load_dotenv

# Загружаем переменные окружения
load_dotenv()

# Supabase Configuration
SUPABASE_URL = os.getenv('SUPABASE_URL', 'https://your-project.supabase.co')
SUPABASE_ANON_KEY = os.getenv('SUPABASE_ANON_KEY', 'your-anon-key')
SUPABASE_SERVICE_ROLE_KEY = os.getenv('SUPABASE_SERVICE_ROLE_KEY', 'your-service-role-key')

# Supabase Storage Configuration
SUPABASE_STORAGE_BUCKET = os.getenv('SUPABASE_STORAGE_BUCKET', 'gtm-assets')

# Database Tables
SUPABASE_TABLES = {
    'users': 'users',
    'artists': 'artists',
    'subscriptions': 'subscriptions',
    'tickets': 'tickets',
    'referrals': 'referrals',
    'giveaways': 'giveaways'
}

# Storage Paths
SUPABASE_STORAGE_PATHS = {
    'avatars': 'avatars',
    'gallery': 'gallery',
    'artists': 'artists',
    'banners': 'banners'
}

def validate_supabase_config():
    """Проверка конфигурации Supabase"""
    required_vars = ['SUPABASE_URL', 'SUPABASE_ANON_KEY', 'SUPABASE_SERVICE_ROLE_KEY']
    missing_vars = [var for var in required_vars if not os.getenv(var)]
    
    if missing_vars:
        print(f"⚠️ Отсутствуют переменные окружения: {missing_vars}")
        return False
    return True

# Экспортируем конфигурацию
__all__ = [
    'SUPABASE_URL', 'SUPABASE_ANON_KEY', 'SUPABASE_SERVICE_ROLE_KEY',
    'SUPABASE_STORAGE_BUCKET', 'SUPABASE_TABLES', 'SUPABASE_STORAGE_PATHS',
    'validate_supabase_config'
] 

File: lib/supabase_config.dart
class SupabaseConfig {
  // Supabase Configuration для проекта rxmtovqxjsvogyywyrha
  static const String supabaseUrl = 'https://rxmtovqxjsvogyywyrha.supabase.co';
  static const String supabaseAnonKey = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InJ4bXRvdnF4anN2b2d5eXd5cmhhIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NTQ1Mjg1NTAsImV4cCI6MjA3MDEwNDU1MH0.gDkJybktFoi486hbIVwppDfmVQlAR0fM4o4Sl1-AxhE';
  
  // Supabase Storage Configuration
  static const String storageBucket = 'gtm-assets';
  
  // Storage Paths
  static const String avatarsPath = 'avatars';
  static const String galleryPath = 'gallery';
  static const String artistsPath = 'artists';
  static const String bannersPath = 'banners';
  
  // API Endpoints
  static const String apiBaseUrl = '$supabaseUrl/rest/v1';
  
  // Database Tables
  static const String usersTable = 'users';
  static const String artistsTable = 'artists';
  static const String subscriptionsTable = 'subscriptions';
  static const String referralsTable = 'referrals';
  static const String giveawaysTable = 'giveaways';
  static const String artistGalleryTable = 'artist_gallery';
  
  // Headers for Supabase API
  static Map<String, String> get headers => {
    'apikey': supabaseAnonKey,
    'Authorization': 'Bearer $supabaseAnonKey',
    'Content-Type': 'application/json',
  };
  
  // Storage URLs
  static String getStorageUrl(String path, String fileName) {
    return '$supabaseUrl/storage/v1/object/public/$storageBucket/$path/$fileName';
  }
  
  // API URLs
  static String getTableUrl(String table) {
    return '$apiBaseUrl/$table';
  }
  
  // Validation
  static bool get isConfigured {
    return supabaseUrl != 'https://your-project.supabase.co' && 
           supabaseAnonKey != 'your-anon-key-here';
  }
  
  // Пример URL для баннера
  static const String citySelectionBannerUrl = 'https://rxmtovqxjsvogyywyrha.supabase.co/storage/v1/object/public/gtm-assets/banners/city_selection_banner.png';
} 